Code/GraphMol/MarvinParse/MarvinWriter.cpp

```diff
@@ -517,6 +520,8 @@ class MarvinCMLWriter {
       }
       for (auto atom : mol->atoms()) {
         auto marvinAtom = new MarvinAtom();
+        marvinMol->pushOwnedAtom(marvinAtom);
+
         marvinMol->atoms.push_back(marvinAtom);
```

Comment:
Is there any reason you would have to do that in two steps?
Why wouldn't pushOwnedAtom also add to atoms?

Reply:
The atoms exist in every mol/sgroup, and many are copies of the atom from its parent or grandparent or some parent about it in the hierarchy.  The ownedAtoms only exist in a couple of mol types - the top level mol or a MarvinSuperatomSgroup.    The purpose of the atoms array is to keep track of which atoms in an sgroup are references to atoms in an ancestor.    The ownedAtoms is to make sure that we keep track of how to destroy them when we are done.  
When a MarvinSuperatomSgroup is expanded in prepartion for creating an RDKit mol, the ownedAtoms are moved from the old MarvinSuperatomSgroup  to its parent.

SO the puposed of the atoms array and the ownedAtoms is very different.  The are several places where an atom (ref) is added to an atoms array and it is already in the correct ownedAtoms array.

So I really would like NOT to move the atoms array addition into the call to pushOwnedAtomUniqPtr.

However, your comments made me think about the two different purposes for atoms and ownedAtoms.  The ownedAtoms are never used to manipulate the data - just to track when to delete them.  Before this PR, I did special work to make sure I new which MOL/Sgroup owned an atom pointer - now I don't.  So there is no reason that ALL atoms could not be owned by the top level MOL.  This is way cleaner - there is no reason to MOVE an owned atom - just create and occasionally release one.   So I am changing the code to do this all atoms are owned by the top level mol.